cli/azd/cmd/middleware/error.go

```diff
@@ -340,3 +365,103 @@ func promptForErrorHandlingConsent(
 
 	return choices[*choiceIndex].Value, nil
 }
+
+// extractSuggestedSolutions extracts the three solutions from the "Brainstorm Solutions" section
+// in the LLM response. Returns a slice of solution strings.
+func extractSuggestedSolutions(llmResponse string) []string {
+	var solutions []string
+
+	// Find the "Brainstorm Solutions" section (case-insensitive, with optional markdown headers)
+	lines := strings.Split(llmResponse, "\n")
+	inSolutionsSection := false
+
+	// Regex to match numbered list items: "1. ", "2. ", "3. "
+	numberPattern := regexp.MustCompile(`^\s*(\d+)\.\s+(.+)$`)
+
+	for _, line := range lines {
+		trimmedLine := strings.TrimSpace(line)
+
+		// Check if we're entering the "Brainstorm Solutions" section
+		if strings.Contains(strings.ToLower(trimmedLine), "brainstorm solutions") {
+			inSolutionsSection = true
+			continue
+		}
+		// If we're in the solutions section, extract numbered items
+		if inSolutionsSection {
+			// Stop if we hit another section header
+			if strings.HasPrefix(trimmedLine, "#") &&
+				!strings.Contains(strings.ToLower(trimmedLine), "brainstorm solutions") {
+				break
+			}
+
+			// Match numbered list items
+			matches := numberPattern.FindStringSubmatch(trimmedLine)
+			if len(matches) == 3 {
+				solutionText := strings.TrimSpace(matches[2])
+				if solutionText != "" {
+					solutions = append(solutions, solutionText)
+				}
+			}
+
+			// Stop after collecting 3 solutions
+			if len(solutions) >= 3 {
+				break
+			}
+		}
+	}
+
+	return solutions
+}
+
+// promptUserForSolution displays extracted solutions to the user and prompts them to select which solution to try.
+// Returns the selected solution text, a flag indicating if user wants to continue with AI fix, and error if any.
+func promptUserForSolution(ctx context.Context, solutions []string, agentName string) (string, bool, error) {
+	choices := make([]*uxlib.SelectChoice, len(solutions)+2)
+
+	// Add the three solutions
+	for i, solution := range solutions {
+		choices[i] = &uxlib.SelectChoice{
+			Value: solution,
+			Label: "Yes. " + solution,
+		}
+	}
+
+	choices[len(solutions)] = &uxlib.SelectChoice{
+		Value: "continue",
+		Label: fmt.Sprintf("Yes, allowing %s to fix the error independently", agentName),
+	}
+
+	choices[len(solutions)+1] = &uxlib.SelectChoice{
+		Value: "cancel",
+		Label: "No, cancel",
+	}
+
+	selector := uxlib.NewSelect(&uxlib.SelectOptions{
+		Message:         fmt.Sprintf("Allow %s to fix the error?", agentName),
+		HelpMessage:     "Select a solution to proceed, continue with AI fix, or cancel",
```

Comment:
This could be a little confusing as the 3 solutions would be fixed with AI as well - thoughts on something like "Select a specific solution or let AI determine the best fix" or "Select a suggested fix, or let AI decide"?